core/iwasm/common/wasm_runtime_common.c

```diff
@@ -887,6 +887,60 @@ wasm_runtime_get_module_package_type(WASMModuleCommon *module)
     return module->module_type;
 }
 
+uint32
+wasm_runtime_get_file_package_version(const uint8 *buf, uint32 size)
+{
+    if (buf && size >= 8) {
+        uint32 version;
+#if (WASM_ENABLE_WORD_ALIGN_READ != 0)
+        uint32 buf32 = *(uint32 *)(buf + sizeof(uint32));
+        buf = (const uint8 *)&buf32;
```

Comment:
It is invalid to read buf[4] to buf[7] below, should read buf[0] to buf[3] instead. How about:
```C
#if (WASM_ENABLE_WORD_ALIGN_READ != 0)
        uint32 buf32 = *(uint32 *)(buf + sizeof(uint32));
        buf = (const uint8 *)&buf32;
        version = buf[0] | buf[1] << 8 | buf[2] << 16 | buf[3] << 24;
#else
        version = buf[4] | buf[5] << 8 | buf[6] << 16 | buf[7] << 24;
#endif
        return version;
```

Reply:
Good catch, I'll fix.